src/components/announcementsList/index.tsx

```diff
@@ -1,65 +1,154 @@
-const NO_ANNOUNCEMENTS_HEADER_COLOR = ORANGE;
-
-const AnnouncementRow = styled(Row)`
-  display: grid;
-  grid-template-columns: 1fr 1fr 1fr;
-  margin-bottom: 20px;
-`;
 const NoAnnouncementsSubText = styled.span`
   display: block;
   text-align: center;
 `;
 
 const NoAnnouncementsText = styled(NoAnnouncementsSubText)`
-  color: ${NO_ANNOUNCEMENTS_HEADER_COLOR};
+  color: ${ORANGE};
   font-size: 36px;
   font-weight: 800;
 `;
 
+const AnnouncementsListWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  justify-content: center;
+  margin-bottom: 40px;
+`;
+
+const PageNumbersWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+`;
+
+const PageNumber = styled.li`
+  font-size: 30px;
+  list-style-type: none;
+  cursor: pointer;
+  padding: 0px 16px;
+  border: 1px solid #d9d9d9;
+  box-sizing: border-box;
+  border-radius: 4px;
+  margin: 0px 8px;
+`;
+
+const SelectedPageNumber = styled(PageNumber)`
+  color: ${LINK};
+  border: 1px solid ${LINK};
+`;
+
+const ArrowButton = styled(PageNumber)`
+  color: #bfbfbf;
+`;
+
+const noAnnouncementsTexts: JSX.Element = (
+  <>
+    <NoAnnouncementsText>{NO_ANNOUNCEMENTS_HEADER}</NoAnnouncementsText>
+    <NoAnnouncementsSubText>
+      {NO_ANNOUNCEMENTS_SUBHEADER}
+    </NoAnnouncementsSubText>
+  </>
+);
+
 export interface AnnouncementsListProps {
   announcements: Announcement[];
 }
+const ANNOUNCEMENTS_PER_PAGE = 6;
 
 const AnnouncementsList: React.FC<AnnouncementsListProps> = ({
   announcements,
 }) => {
-  const [numRows, setNumRows] = useState(0);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handleNextPage = () => {
+    setCurrentPage((page) => page + 1);
+  };
 
-  useEffect(() => {
-    const rows = Math.ceil(announcements.length / COLUMNS_PER_ROW);
-    setNumRows(rows);
-  }, [announcements]);
+  const handlePreviousPage = () => {
+    setCurrentPage((page) => page - 1);
+  };
+
+  const handleNoOnClick = () => {
+    setCurrentPage((page) => page);
+  };
+
+  const handlePageClick = (pageNum: number) => {
+    setCurrentPage(() => pageNum);
+  };
+
+  const noAnnouncements: boolean = announcements.length === 0;
+  const indexOfLastAnnouncement: number = currentPage * ANNOUNCEMENTS_PER_PAGE;
+  const indexOfFirstAnnouncement: number =
+    indexOfLastAnnouncement - ANNOUNCEMENTS_PER_PAGE;
+  const currentAnnouncements: Announcement[] = announcements.slice(
+    indexOfFirstAnnouncement,
+    indexOfLastAnnouncement,
+  );
+
+  const lastPage: number =
+    Math.ceil(announcements.length / ANNOUNCEMENTS_PER_PAGE) + 1;
+  const renderPageNumbers: JSX.Element[] = range(1, lastPage).map(
+    (num: number) =>
+      num !== currentPage ? (
+        <PageNumber key={num} value={num} onClick={() => handlePageClick(num)}>
+          {num}
+        </PageNumber>
+      ) : (
+        <SelectedPageNumber key={num} value={num}>
+          {num}
+        </SelectedPageNumber>
+      ),
+  );
```

Comment:
This is pretty, I like. I think because it's only used one place in the below return statement, we should just do this inline.

Reply:
That makes sense